pallets/credit/src/lib.rs

```diff
@@ -728,17 +837,64 @@ pub mod pallet {
             if new_id.is_none() {
                 return false;
             }
+            let new_id = new_id.unwrap();
 
-            old_data.campaign_id = new_id.unwrap();
-            old_data.reward_eras += 180;
+            if old_data.campaign_id == new_id {
+                old_data.reward_eras += 180;
+            } else {
+                old_data.campaign_id = new_id;
+                old_data.reward_eras = DEFAULT_REWARD_ERAS;
+            }
 
-            UserCredit::<T>::insert(who, old_data.clone());
+            UserCredit::<T>::insert(who, old_data);
             Self::update_credit_history(who, expire_era);
             true
         }
+
+        fn do_add_credit(account_id: T::AccountId, credit_data: CreditData) -> DispatchResult {
```

Comment:
This method won't return `Err`, how about not return a `Result`

Reply:
good